hadoop/Dockerfile

```diff
@@ -153,5 +154,20 @@ ENV ASYNC_PROFILER_HOME=/stackable/async-profiler
 ENV HADOOP_YARN_HOME=/stackable/hadoop
 ENV HADOOP_MAPRED_HOME=/stackable/hadoop
 
+# Remove unneeded binaries:
+#  - source code
+#  - mapreduce/yarn binaries that were built as cross-project dependencies
+#  - minicluster (only used for testing) and test .jars
+#  - json-io: this is a transitive dependency pulled in by cedarsoft/java-utils/json-io and is excluded in 3.4.0. See CVE-2023-34610.
+RUN rm -rf /stackable/hadoop/share/hadoop/common/sources/ && \
+  rm -rf /stackable/hadoop/share/hadoop/hdfs/sources/ && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/sources/ && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/lib/json-io-*.jar && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/lib/hadoop-mapreduce-client-*.jar && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/lib/hadoop-yarn-server*.jar && \
```

Comment:
Shouldn't we exclude these from building instead?

Reply:
We exclude the modules from the final build, but all the jars are built regardless of the `mvn .... -pl` directive, due to cross-dependencies. Hence the second step to strip them here.

Reply:
it looks like `-pl '!hadoop-tools/hadoop-pipes,!hadoop-yarn-project,!hadoop-mapreduce-project,!hadoop-minicluster'` should exclude at least some of these, maybe it's not working? maybe they cannot be excluded? A comment would be good in that case, explaining why they cannot be excluded from the build.

Reply:
-pl results in there being e.g. no share/hadoop/yarn or share/hadoop/mapreduce folders in the image, although various yarn/MR binaries are still needed in e.g. share/tools. I'll add a comment.

Reply:
Comment is hopefully a bit clearer here: 878cc24

Reply:
Works for me